app/Http/Controllers/PDFController.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use DOMPDF;
+use LibreNMS\Config;
+use Illuminate\Support\Facades\Route;
+use LibreNMS\Util\DynamicConfig;
+
+require '/opt/librenms/includes/init.php';
+require_once '/opt/librenms/includes/html/functions.inc.php';
+
+class PDFController extends Controller
+{
+	public function preview(DynamicConfig $config, $path, Request $request){
+		$clean_path = \LibreNMS\Util\Clean::fileName($path);
+		$view 	= 'pdf.' . $clean_path;
+		
+		// this is the URL used in the button "Create PDF" - probably can be done another way but this works.... 
+		$route	= '/dompdf/generate/'. $clean_path;
+
+		$header_logo = \LibreNMS\Config::get('dompdf.header_logo');
+		$footer_logo = \LibreNMS\Config::get('dompdf.footer_logo');
+		
+		// Validate that the view exists 
+		if (view()->exists($view)) {
+			// Run report and return JSON object - saved to $json
+			include_once "/opt/librenms/includes/html/reports/$clean_path.pdf.inc.php";
```

Comment:
not allowed

Reply:
should I run the Database query here instead?